app/api/ada/lib/utils.js

```diff
@@ -54,3 +64,142 @@ const _getTxCondition = (state: string): AdaTransactionCondition => {
   if (state === 'Pending') return 'CPtxApplying';
   return 'CPtxWontApply';
 };
+
+/** Repeatedly scan addresses until there is a contiguous block of `scanSize` addresses unused
+ * @returns all scanned addresses
+ */
+export async function discoverAllAddressesFrom(
```

Comment:
I don’t think `utils` is the best place to put this code. Utils is more for generic stuff like `localeDate`. All the code related to the Ada should be somewhere else.

Reply:
Yeah I don't understand the motivation for copying all of this to a new file. What was wrong with leaving it in `restoreAdaWallet.js`?

Reply:
![image](https://user-images.githubusercontent.com/19986226/49353631-1d65b480-f702-11e8-902f-eaab322de14d.png)

Its repeated in both `restore wallet` and `create trezor wallet` process

Reply:
DONE